crates/driver/src/infra/api/solve/dto/auction.rs

```diff
@@ -130,12 +132,16 @@ pub enum Error {
     InvalidAuctionId,
     #[error("surplus fee is missing for limit order")]
     MissingSurplusFee,
+    #[error("received an auction with an exceeded deadline")]
+    DeadlineExceeded,
 }
 
 #[serde_as]
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Auction {
+    // TODO This should not be optional. Also, this seems to indicate that there should be two
+    // different auction types, or maybe set the ID to Default::default() for quoting?
```

Comment:
I think we purposefully wanted to use the same API for solving and quoting to reduce the additional work on the `solver engine`.